File: examples/export-pages-and-assets/rest/ruby/images.rb
# Exports a page and all its linked resources (CSS, JS, images) as a ZIP file
# using the Browserless /export endpoint.
# Uses net/http and json from the standard library — no gems required.
#
# Run: ruby images.rb

require 'net/http'
require 'json'
require 'uri'

TOKEN = 'YOUR_API_TOKEN_HERE'

uri = URI("https://production-sfo.browserless.io/export?token=#{TOKEN}")
http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = true

request = Net::HTTP::Post.new(uri)
request['Content-Type'] = 'application/json'
request.body = JSON.generate({ url: 'https://example.com', includeResources: true })

response = http.request(request)

File.binwrite('page.zip', response.body)
puts 'Saved page.zip'

File: examples/export-pages-and-assets/rest/php/images.php
<?php
// Exports a page and all its linked resources (CSS, JS, images) as a ZIP file
// using the Browserless /export endpoint.
// Uses PHP's built-in curl — no Composer packages needed.
//
// Run: php images.php

$token = 'YOUR_API_TOKEN_HERE';

$ch = curl_init('https://production-sfo.browserless.io/export?token=' . $token);
curl_setopt_array($ch, [
    CURLOPT_POST           => true,
    CURLOPT_HTTPHEADER     => ['Content-Type: application/json'],
    CURLOPT_POSTFIELDS     => json_encode([
        'url'              => 'https://scraping-sandbox.netlify.app/harvest-direct',
        'includeResources' => true,
    ]),
    CURLOPT_RETURNTRANSFER => true,
]);
$body = curl_exec($ch);
curl_close($ch);

file_put_contents('page.zip', $body);
echo "Saved page.zip\n";
